metaserver/src/main/scala/scala/meta/languageserver/compiler/CompletionProvider.scala

```diff
@@ -41,17 +62,28 @@ object CompletionProvider extends LazyLogging {
     val position = unit.position(cursor.offset)
     val isUsedLabel = mutable.Set.empty[String]
     val items = List.newBuilder[CompletionItem]
-    safeCompletionsAt(compiler, position).foreach { r =>
-      val label = r.symNameDropLocal.decoded
-      if (!isUsedLabel(label)) {
-        isUsedLabel += label
-        items += CompletionItem(
-          label = label,
-          detail = Some(r.sym.signatureString),
-          kind = Some(completionItemKind(r))
-        )
+    safeCompletionsAt(compiler, position)
+      .sortBy {
+        case TypeMember(sym, _, true, inherited, viaView) =>
+          // logger.debug(s"Relevance of ${sym.name}: ${computeRelevance(sym, viaView, inherited)}")
+          (-computeRelevance(sym, viaView, inherited), sym.nameString)
+        case ScopeMember(sym, _, true, _) =>
+          (-computeRelevance(sym, NoSymbol, false), sym.nameString)
+        case r => (0, r.sym.nameString)
+      }
+      .zipWithIndex
+      .foreach { case (r, idx) =>
+        val label = r.symNameDropLocal.decoded
+        if (!isUsedLabel(label)) {
+          isUsedLabel += label
+          items += CompletionItem(
+            label = label,
+            detail = Some(r.sym.signatureString),
+            kind = Some(completionItemKind(r)),
+            sortText = Some(f"${idx}%05d")
```

Comment:
sorting is lexicographic, so we can't just do `idx.toString` because that would lead to `0, 1, 10, 2...`.